File: LeetCode/Go/102-tree-二叉树的层序遍历.go
/*
 * @lc app=leetcode.cn id=102 lang=golang
 *
 * [102] 二叉树的层序遍历
 */
// package leetcode

// @lc code=start
func levelOrder(root *TreeNode) [][]int {
	var res = make([][]int, 0)

	var level func(*TreeNode, int)
	level = func (root *TreeNode, l int)  {
		if root == nil {
			return
		}
		if l >= len(res) {
			res = append(res, []int{})
		}
		res[l] = append(res[l], root.Val)
		level(root.Left, l + 1)
		level(root.Right, l + 1)
	}

	level(root, 0)
	return res
}
// @lc code=end

File: LeetCode/Java/102-tree-二叉树的层序遍历.java
import java.util.*;

/*
 * @lc app=leetcode.cn id=102 lang=java
 *
 * [102] 二叉树的层序遍历
 */

// @lc code=start
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        level(root, 0, res);
        return res;
    }

    private void level(TreeNode root, int step, List<List<Integer>> res) {
        if (root == null)
            return;
        if (res.size() == step)
            res.add(new ArrayList<>());
        res.get(step).add(root.val);
        level(root.left, step + 1, res);
        level(root.right, step + 1, res);
    }
}
// @lc code=end
